lib/operations/connect.js

```diff
@@ -245,6 +247,8 @@ function connect(mongoClient, url, options, callback) {
     if (_finalOptions.socketTimeoutMS == null) _finalOptions.socketTimeoutMS = 360000;
     if (_finalOptions.connectTimeoutMS == null) _finalOptions.connectTimeoutMS = 30000;
     if (_finalOptions.retryWrites == null) _finalOptions.retryWrites = true;
+    if (_finalOptions.useRecoveryToken == null) _finalOptions.useRecoveryToken = true;
+    if (_finalOptions.readPreference == null) _finalOptions.readPreference = 'primary';
```

Comment:
Would it be better to have this set in `startTransaction`? Is that where the test was failing?

Reply:
I think that might be better. My feeling here is that we might be surfacing a bug that was not present..

Reply:
Actually, looking into the specs, it says [here](https://github.com/mongodb/specifications/blob/master/source/driver-read-preferences.rst#read-preferences) that the default value for a readPreference is `primary` in a MongoClient. The reason the transaction test is failing is b/c it has no specified value, and is attempting to inherit the value from the MongoClient. So I think the current solution is proper.